src/java/picard/sam/util/ReadNameParsingUtils.java

```diff
@@ -29,24 +29,29 @@
 public class ReadNameParsingUtils {
 
     /**
-     * Single pass method to parse the read name for the default regex.  This will only insert the 2nd to the 4th
-     * tokens (inclusive).  It will also stop after the fifth token has been successfully parsed.
+     * Single pass method to parse the read name for the default regex.  Examines the last three fields as split by the delimiter.
      */
     public static int getRapidDefaultReadNameRegexSplit(final String readName, final char delim, final int[] tokens) {
```

Comment:
Perhaps this ought to be renamed to just "getLastThreeFIelds" ?  It's not clear that it's all that related to the default regex any more, other than that in the specific case of Illumina read names, it will mostly function the same way.